android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -388,116 +427,292 @@ private void reLayout(View view) {
+                    Log.e("ExoPlayer Exception", ex.toString());
+                    eventEmitter.error(ex.toString(), ex, "1001");
                 }
-
-                // Initializing the playerControlView
-                initializePlayerControl();
-                setControls(controls);
-                applyModifiers();
             }
         }, 1);
+        
+    }
+
+    private void initializePlayerCore(ReactExoplayerView self) {
+        ExoTrackSelection.Factory videoTrackSelectionFactory = new AdaptiveTrackSelection.Factory();
+        self.trackSelector = new DefaultTrackSelector(videoTrackSelectionFactory);
+        self.trackSelector.setParameters(trackSelector.buildUponParameters()
+                .setMaxVideoBitrate(maxBitRate == 0 ? Integer.MAX_VALUE : maxBitRate));
+
+        DefaultAllocator allocator = new DefaultAllocator(true, C.DEFAULT_BUFFER_SEGMENT_SIZE);
+        RNVLoadControl loadControl = new RNVLoadControl(
+                allocator,
+                minBufferMs,
+                maxBufferMs,
+                bufferForPlaybackMs,
+                bufferForPlaybackAfterRebufferMs,
+                -1,
+                true,
+                backBufferDurationMs,
+                DefaultLoadControl.DEFAULT_RETAIN_BACK_BUFFER_FROM_KEYFRAME
+        );
+        DefaultRenderersFactory renderersFactory =
+                new DefaultRenderersFactory(getContext())
+                        .setExtensionRendererMode(DefaultRenderersFactory.EXTENSION_RENDERER_MODE_OFF);
+        player = new SimpleExoPlayer.Builder(getContext(), renderersFactory)
+                    .setTrackSelector​(self.trackSelector)
+                    .setBandwidthMeter(bandwidthMeter)
+                    .setLoadControl(loadControl)
+                    .build();
+        player.addListener(self);
+        player.addMetadataOutput(self);
+        exoPlayerView.setPlayer(player);
+        audioBecomingNoisyReceiver.setListener(self);
+        bandwidthMeter.addEventListener(new Handler(), self);
+        setPlayWhenReady(!isPaused);
+        playerNeedsSource = true;
+
+        PlaybackParameters params = new PlaybackParameters(rate, 1f);
+        player.setPlaybackParameters(params);
+    }
+
+    private DrmSessionManager initializePlayerDrm(ReactExoplayerView self) {
+        DrmSessionManager drmSessionManager = null;
+        if (self.drmUUID != null) {
+            try {
+                drmSessionManager = self.buildDrmSessionManager(self.drmUUID, self.drmLicenseUrl,
+                        self.drmLicenseHeader);
+            } catch (UnsupportedDrmException e) {
+                int errorStringId = Util.SDK_INT < 18 ? R.string.error_drm_not_supported
+                        : (e.reason == UnsupportedDrmException.REASON_UNSUPPORTED_SCHEME
+                        ? R.string.error_drm_unsupported_scheme : R.string.error_drm_unknown);
+                eventEmitter.error(getResources().getString(errorStringId), e, "3003");
```

Comment:
This is a verry good thing to unify errors, but:
- I think you should remove the getResources().getString() from the api ...
   There is no reason to have translated error string coming from a library because UI will not be able to handle it depending of the platform language. 
   The translated value will never be alligned with customer requierments.

Please also add documentation of error codes, then it should be unified with ios / windows errors and this onError API will be good

Reply:
here is onError definition I use in my fork:

#### onError
Callback function that is called when the player fails due to an error.

Payload:

| Property                 | Type    | Description                                                 |
| ------------------------ | ------- | ----------------------------------------------------------- |
| code | string | String describing the error code |
| exception | string | String detailing the error (for debugging) |


| errorString                 | Type    |
| ------------------------ | ------- | ----------------------------------------------------------- |
| Unknown | unexpected error |
| DRM | DRM initialization issue |
| Decoder | Video decoder issue |
| Format | Format not supported |
| InvalidState | An invalid action has been requested |
| Canceled | Request has been canceled |
| Network | Network access issue |
| InvalidParamUrl | Source is not valid |
| InvalidParam | An invalid paramter has been provided |
| Timeout | Timeout during initialization |
| MaxConcurentStream | Maximum simultaneous stream count has been reach |
| Restricted | Content cannot be played due to restriction (geolocation for exemple) |
| DeviceNotInHousehold | Cannot play due to Account configuration issue |
| ContentNotFound | asset not found |
| Specific to WebView |
| Html5 | HTML5 player Error |
| VideoNotPlayableInEmbdedPlayer | content cannot be played in webview |
| WebViewNotAvailable | Webview not available on device |

Example:

```
{
  error:
  {
    errorString: 'Network',
    errorException:  'Yeah I had an error, that is bad',
  }
}
```

Reply:
I think this is enough to handle error on UI application